src/appleseed/foundation/image/conversion.cpp

```diff
@@ -0,0 +1,150 @@
+
+//
+// This source file is part of appleseed.
+// Visit https://appleseedhq.net/ for additional information and resources.
+//
+// This software is released under the MIT license.
+//
+// Copyright (c) 2019 Francois Beaune, The appleseedhq Organization
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files (the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+//
+
+// Interface header.
+#include "conversion.h"
+
+// appleseed.foundation headers.
+#include "foundation/image/canvasproperties.h"
+#include "foundation/image/color.h"
+#include "foundation/image/colorspace.h"
+#include "foundation/image/image.h"
+#include "foundation/utility/string.h"
+
+// Standard headers.
+#include <cassert>
+#include <cstddef>
+
+using namespace std;
+namespace bf = boost::filesystem;
+
+namespace foundation
+{
+
+bool is_linear_image_file_format(const bf::path& path)
+{
+    const string extension = lower_case(path.extension().string());
+
+    return
+        extension == ".exr"  ||
+        extension == ".tiff" ||
```

Comment:
We cannot assume that TIFFs contain floating point pixels. Most do not.
Checking in Frame, we are saving linear 16 bits TIFFs, which is wrong. 
Images with integer pixels should be sRGB (I opened an issue to track that).

Reply:
Ok.

Reply:
I'm going to be pedantic here, but you **do** have enough precision with uint16 TIFFs to have a smooth tonal progression without banding ([Barten threshold](https://www.mysterybox.us/blog/2016/10/19/hdr-video-part-3-hdr-video-terms-explained)), though using a perceptual encoding would make better use of the format.